homeassistant/components/tplink/switch.py

```diff
@@ -29,15 +32,25 @@
 
 async def async_setup_entry(hass, config_entry, async_add_entities):
     """Set up discovered switches."""
-    devs = []
-    for dev in hass.data[TPLINK_DOMAIN][CONF_SWITCH]:
-        devs.append(SmartPlugSwitch(dev))
-
-    async_add_entities(devs, True)
+    async_add_entities_retry(
+        hass,
+        async_add_entities,
+        hass.data[TPLINK_DOMAIN][CONF_SWITCH],
+        add_entity
+    )
 
     return True
 
 
+def add_entity(device: SmartPlug, async_add_entities):
+    """Check if device is online and add the entity."""
+    device.get_sysinfo()
+    async_add_entities(
+        [SmartPlugSwitch(device)],
+        True
```

Comment:
Please use keyword arguments for booleans, it will make it easier for anyone reading this code to understand the meaning of "True".

Reply:
Done